src/Emitter/EmitterStack.php

```diff
@@ -45,56 +56,29 @@ public function emit(ResponseInterface $response) : bool
         return false;
     }
 
-    /**
-     * Set an emitter on the stack by index.
-     *
-     * @param mixed $index
-     * @param EmitterInterface $emitter
-     * @return void
-     * @throws InvalidArgumentException if not an EmitterInterface instance
-     */
-    public function offsetSet($index, $emitter)
+    public function offsetSet(int $index, EmitterInterface $emitter): void
```

Comment:
`SplStack` implements a few interfaces that make this and other methods to integrate more easily to the language. This change is then a BC-break: https://3v4l.org/rNtHA#v8.0.8 vs https://3v4l.org/ZgTsb#v8.0.8

To solve that, we'd need to implement `Iterator`, `ArrayAccess`, and `Countable`: https://3v4l.org/CuTCA#v8.0.8

This also means we must keep the parameters covariant (`ArrayAccess` defines `$offset` and `$value` as `mixed`).

Reply:
I've rollbacked this and added proper annotations instead.

Reply:
BTW: targeting 2.0 as a new major and thus BC breaks should be fine. But I guess that we can also live with some psalm suppressions 👍🏼